src/vals.c

```diff
@@ -71,22 +61,41 @@ static void vals_compact(struct vals *vals)
     vals->len = j + 1;
 }
 
-static struct vals *vals_from_pairs(struct rill_pairs *pairs)
+static struct vals *from_pairs(struct rill_pairs *pairs, enum rill_col col)
 {
     struct vals *vals =
         calloc(1, sizeof(*vals) + sizeof(vals->data[0]) * pairs->len);
+
     if (!vals) return NULL;
 
     vals->len = pairs->len;
-    for (size_t i = 0; i < pairs->len; ++i)
-        vals->data[i] = pairs->data[i].val;
+    if (col == rill_col_a) {
```

Comment:
Might be a bit cleaner if you added a function to pairs called `rill_pairs_col(pairs, rill_col_a)`. That way you don't have to add forks in your code when dealing with a specific column.

Reply:
Fixed.